documentation/add/09_Serializable_Interfaces.md

```diff
@@ -0,0 +1,132 @@
+| `obj = from_bare_struct(obj,data)` | Reverse of `to_bare_struct` method. Sets the values of the public properties of the serializable interface from the structure, obtained from `to_bare_struct` method |
+| `data = to_struct(obj)` | converts `obj` or array of `obj` into the `data` structure using `to_bare_struct` method and adds complimentary information about the class name and the size and shape of the `obj` array if relevant. |
+| *Static:* | |
+| `obj = from_struct(data)` | Reverse of `to_struct` method, recovering the object or array of objects from the `data`, generated by the `to_struct` method |
+
+Standard `loadobj/saveobj` and `serialize/deserialize` methods use the data obtained from the methods described in **Table 2** or the methods themselves to provide/use input for/from Matlab save/load methods or serialize/deserialize methods to save/load data to/from `.mat` files or transform the objects from/to linear arrays of bytes for communications over serial pipes. The `serialize/deserialize` routines use Herbert-defined `serialize/deserialize` methods over the structures obtained using `to_struct/from_struct` methods.
+
+Two remaining methods presented on the **Fig 1** namely static `loadobj` and `from_old_struct` methods need overloading if/when you want to support loading the versions of a class, different from the the current class version. By default, a standard `loadobj` method exists and this method expects to receive the structure, produced by `to_struct` method. `to_struct` method adds information about the class name, so generic `loadobj` extracts this information and recovers the class. If this information is missing, which may happen when you loading a class structure, generated before the object become `serializable`, you need to overload `loadobj` to provide the empty instance of the class, as described by the code snippet below:
+```Matlab
+        methods(Static)
+            function obj = loadobj(S)
+                % boilerplate loadobj method, calling generic method of
+                % saveable class. Put it as it is replacing the
+                "ChildClass" name by the name of the class the loadobj is
+                the method of.
+                obj = ChildClass();
+                obj = loadobj@serializable(S,obj);
+            end
+        end
+```
+By default, the method `from_old_struct` calls `from_bare_struct` method, so it does not need overloading if the structure of your class public interface have not changed.
+If it have changed, `from_old_struct` needs overloading with the code, converting the old information, stored in the previous class structure into the information, necessary to define new class.
+
+## Interdependent properties problem.
+
+All serializable interface described above and used inside the code is build under assumption that you define class value by assigning the values of the properties one by one. 
```

Comment:
I think we need more than this as from_old_struct has to consider a variety of older version. Some examples would be good. But it is probably not needed immediately

Reply:
This is ADD after all  -- ideas, not detailed instruction. I would refer to existing code for examples.